Mathlib/Probability/Distributions/Gamma.lean

```diff
@@ -0,0 +1,301 @@
+  have leftSide : ∫⁻ x in Iio 0, gammaPdf a r x = 0 := by
+    rw [set_lintegral_congr_fun measurableSet_Iio
+      (ae_of_all _ (fun x (hx : x < 0) ↦ gammaPdf_of_neg hx)), lintegral_zero]
+  have rightSide :
+      ∫⁻ x in Ici 0, gammaPdf a r x =
+      ∫⁻ x in Ici 0, ENNReal.ofReal (r ^ a / (Gamma a) * x ^ (a-1) * exp (-(r * x))) :=
+    set_lintegral_congr_fun isClosed_Ici.measurableSet
+      (ae_of_all _ (fun x (hx : 0 ≤ x) ↦ gammaPdf_of_nonneg hx))
+  simp only [leftSide, add_zero]
+  rw [rightSide]
+  rw [← integral_eq_lintegral_of_nonneg_ae]
+  · rw [MeasureTheory.integral_Ici_eq_integral_Ioi, pow_exp_integral_Ioi_insert_1 ha hr]
+  · unfold EventuallyLE
+    simp only [Pi.zero_apply]
+    rw [ae_restrict_iff]
+    simp only [mem_Ici]
+    · apply ae_of_all
+      intro x hx
+      positivity
+    · refine measurableSet_le measurable_const ?_
+      refine Measurable.mul ?_ ?_
+      · measurability
+      · apply (Measurable.neg ?_).exp
+        exact measurable_id.const_mul _
+  · rw [← restrict_Ioi_eq_restrict_Ici]
+    refine ContinuousOn.aestronglyMeasurable ?_ measurableSet_Ioi
+    refine ContinuousOn.mul ?_ (Continuous.continuousOn (by continuity))
+    exact continuousOn_const.mul (continuousOn_id.rpow_const fun x hx ↦ Or.inl (ne_of_gt hx))
+
+end GammaPdf
+
+open MeasureTheory
+
+/-- Measure defined by the Gamma Distribution -/
+noncomputable
+def gammaMeasure (a r : ℝ) : Measure ℝ :=
+  volume.withDensity (gammaPdf a r)
+
+instance instIsProbabilityMeasureGamma (a r : ℝ) (ha : 0 < a) (hr : 0 < r) :
+    IsProbabilityMeasure (gammaMeasure a r) where
+  measure_univ := by simp [gammaMeasure, lintegral_gammaPdf_eq_one a r ha hr]
+
+section GammaCdf
+
+/-- CDF of the Gamma Distribution -/
+noncomputable
+def gammaCdfReal (a r : ℝ) : StieltjesFunction :=
+  cdf (gammaMeasure a r)
+
+lemma gammaCdfReal_eq_integral (a r x : ℝ) (ha : 0 < a) (hr : 0 < r) :
+    gammaCdfReal a r x = ∫ x in Iic x, gammaPdfReal a r x := by
+  have : IsProbabilityMeasure (gammaMeasure a r) := instIsProbabilityMeasureGamma a r ha hr
+  rw [gammaCdfReal, cdf_eq_toReal, gammaMeasure, withDensity_apply _ measurableSet_Iic]
+  refine (integral_eq_lintegral_of_nonneg_ae ?_ ?_).symm
+  · exact ae_of_all _ fun b ↦ by simp [Pi.zero_apply, gammaPdfReal_nonneg ha hr]
+  · exact (Measurable.aestronglyMeasurable (measurable_gammaPdfReal a r)).restrict
+
+lemma gammaCdfReal_eq_lintegral (a r x : ℝ) (ha : 0 < a) (hr : 0 < r) :
+    gammaCdfReal a r x = ENNReal.toReal (∫⁻ x in Iic x, gammaPdf a r x) := by
+  have : IsProbabilityMeasure (gammaMeasure a r) := instIsProbabilityMeasureGamma a r ha hr
+  simp only [gammaPdf, gammaCdfReal, cdf_eq_toReal]
+  simp only [gammaMeasure, measurableSet_Iic, withDensity_apply, gammaPdf]
+
+end GammaCdf
+
+open Complex
+
+/-- Expresses the integral over Ioi 0 of t^(a-1) * exp(-r*t) in terms of the Gamma function,
+for complex a. -/
+lemma pow_exp_integral_Ioi_complex {a : ℂ} {r : ℝ} (ha : 0 < a.re) (hr : 0 < r) :
```

Comment:
Looks great, thanks for adding this! Can you maybe move it to `SpecialFunctions.Gamma.Basic` or somewhere like that (somewhere shortly after the definition of `Complex.Gamma`)?

I would also suggest a different name, maybe `integral_cpow_mul_exp_neg_mul_Ioi`, for this lemma; and similarly the real version should be `integral_rpow_mul_exp_neg_mul_Ioi`. (Generally we distinguish between different kinds of exponentiation depending on the type of the exponent, so `pow` means Nat exponent, and other kinds of powers are `zpow, rpow, cpow`.)

How painful would it be to deduce the real-variable version (the current `pow_exp_integral_Ioi`) from the complex version, rather than giving a separate proof as currently?

Reply:
Thank you, I'll incorporate all this and make a new commit right away. Deducing should be easy, but I wanted to wait until the location was determined, so I wouldn't have to keep changing imports in Distributions/Gamma.

Reply:
I'm having some issues with the coercions to give the real-variable version directly from the complex version, it is in this commit (which probably won't build because of the 'sorry')

Reply:
New commit made, all points are now reflected there!